### ISSUE TITLE
Container issues in the docker driven execution (-dockerize) 

### ISSUE THREAD

**Issue by mribeirodantas** [nextflow-io/nextflow#3137]
## Bug report 

The option `-dockerize` is not working at all.

### Expected behavior and actual behavior

If Nextflow is not in the `$PATH`, and thus you're running the -dockerize option like `./path/nextflow -dockerize run hello`, it won't work. You'll only get:

`docker: invalid reference format.
See 'docker run --help'.`

If nextflow is in the `$PATH`, it will download the layers of the image, build the image but the container will throw several errors, such as:
```
/usr/local/bin/entry.sh: line 46: addgroup: command not found
/usr/local/bin/entry.sh: line 47: adduser: command not found
chown: invalid user: 'nextflow'
/usr/local/bin/entry.sh: line 50: su: command not found
```

In Apple Silicon, or the following in CentOS 9 (ARM, virtualized in a machine with Apple Silicon):

```
exec /usr/local/bin/entry.sh: exec format error
```

@bentsherman tried on Windows (with WSL) and it also didn't work.

### Steps to reproduce the problem

`nextflow -dockerize run hello`

### Program output 

There is no output but the one above from the errors in the container.

### Environment 

* Nextflow version: 22.04.5 build 5708
* Java version: openjdk 17.0.3 2022-04-19 (Apple Silicon), openjdk 11.0.16 2022-07-19 (CentOS 9, ARM)
* Operating system: macOS Monterey 12.5.1, and CentOS9 (ARM)
* Bash version: zsh 5.8.1 (x86_64-apple-darwin21.0) and GNU Bash 5.1.8(1) in CentOS9 (ARM)

**Comment by juanesarango** [comment]
Is there any workaround to use this feature that hasn't been released yet?
Thanks

**Comment by bentsherman** [comment]
This issue was fixed, the PR just wasn't linked properly to close it. The fix is available since [v22.11.0-edge](https://github.com/nextflow-io/nextflow/releases/tag/v22.11.0-edge).

**Comment by juanesarango** [comment]
It works indeed. Thanks! 🙌🏽 

```bash
$ export NXF_VER=22.11.0-edge
$ nextflow -dockerize run hello

Unable to find image 'nextflow/nextflow:22.11.0-edge' locally
22.11.0-edge: Pulling from nextflow/nextflow
68028ec3b506: Pull complete
d137d2313065: Pull complete
de4a3a4453a7: Pull complete
f643153b3a20: Pull complete
f88c1675f882: Pull complete
c3be39b40ed3: Pull complete
c4ca585dfe65: Pull complete
Digest: sha256:58d56cbe6fce893637421c3e2b491084c11508a36686a4b3ea3a1c37f38f0b70
Status: Downloaded newer image for nextflow/nextflow:22.11.0-edge

N E X T F L O W  ~  version 22.11.0-edge
Pulling nextflow-io/hello ...
 downloaded from https://github.com/nextflow-io/hello.git
Launching `https://github.com/nextflow-io/hello` [magical_goldberg] DSL2 - revision: 4eab81bd42 [master]
[3c/f1b839] Submitted process > sayHello (4)
[d4/c2a32e] Submitted process > sayHello (1)
[64/ad17ee] Submitted process > sayHello (2)
[8f/3829ea] Submitted process > sayHello (3)
Hola world!

Bonjour world!

Hello world!

Ciao world!
```
